streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java

```diff
@@ -654,6 +655,10 @@ public void init(final ProcessorContext context) {
                     public KeyValue<Long, Long> transform(final Long key, final Long value) {
                         if (gcInjected.compareAndSet(true, false)) {
                             while (doGC) {
+                                final StreamThread thread = (StreamThread) Thread.currentThread();
+                                if (thread.isInterrupted() || !thread.isRunning()) {
```

Comment:
In `KafkaStreams.close`, we don't interrupt the StreamThreads but set them to `state:=PENDING_SHUTDOWN`. I'm checking both here, though, because the thread might be interrupted for other reasons, though, and it's generally polite to make sure you haven't been interrupted if you're blocking.